hs_core/views/resource_access_api.py

```diff
@@ -0,0 +1,182 @@
+
+    class Meta:
+        model = GroupResourcePrivilege
+        fields = ('id', 'privilege', 'group', 'resource', 'grantor')
+
+
+class UserResourcePrivilegeSerializer(serializers.ModelSerializer):
+    privilege = PrivilegeField()
+
+    class Meta:
+        model = UserResourcePrivilege
+        fields = ('id', 'privilege', 'user', 'resource', 'grantor')
+
+
+class ResourceAccessUpdateDelete(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Read, update, or delete a resource
+
+    REST URL: hsapi/resource/{pk}/access
+    HTTP method: GET
+    :return: (on success): JSON representation of resource access with 'groups' and 'users' keys.
+
+    REST URL: hsapi/resource/{pk}/access?(user_id=#|group_id=#)
+    HTTP method: DELETE
+
+    :type str
+    :param user_id: user ID to remove
+    :type str
+    :param group_id: group ID to remove
+    :return: (on success): Success or Error JSON object
+
+    REST URL: hsapi/resource/{pk}/access
+    HTTP method: PUT
+    :return: (on success): Success or Error JSON object
+
+    :type str
+    :param user_id: user ID to remove
+    :type str
+    :param group_id: group ID to remove
+    :type PrivilegeCode int
+    :param privilege: PrivilegeCode to specifiy access level
+    :return: (on success): Success or Error JSON objectit
+    """
+
+    serializer_class = UserResourcePrivilegeSerializer
+
+    allowed_methods = ('GET', 'PUT', 'DELETE')
+
+    def get(self, request, pk):
+        view_utils.authorize(request, pk,
+                             needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE_ACCESS)
+
+        user_resource_serializer, group_resource_serializer = self.get_serializer_classes()
+        user_resource_privilege, group_resource_privilege = self.get_queryset(pk, request.user)
+
+        response_data = dict()
+        response_data['users'] = \
+            user_resource_serializer(user_resource_privilege, many=True).data
+        response_data['groups'] = \
+            group_resource_serializer(group_resource_privilege, many=True).data
+
+        return Response(data=response_data, status=status.HTTP_200_OK)
+
+    def put(self, request, pk):
+        view_utils.authorize(request, pk,
+                             needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE_ACCESS)
+        user_access = UserAccess(user=request.user)
+        resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
+        keys = request.data.keys()
+
+        if "user_id" in keys and "group_id" in keys:
+            return Response(
+                data={
+                    'error': "Request cannot contain both a 'user_id' and a 'group_id' parameter."
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        if "user_id" in keys and "privilege" in keys:
+            user_to_add = utils.user_from_id(request.data['user_id'])
+            user_access.share_resource_with_user(resource, user_to_add, request.data['privilege'])
```

Comment:
Should this also be within ```try ... except```?

Also should we be validating the value in ```request.data['privilege']``` before using it?